File: node_version/modules/date_time.js

export default class DateTime{
    constructor(date_time = new Date()){
        this.time = date_time;
        this.year = date_time.getFullYear();
        this.mon = date_time.getMonth() + 1;
        this.day = date_time.getDate();
        this.hour = date_time.getHours();
        this.min = date_time.getMinutes();
        this.digital = this.min + this.hour*100 + this.day*10000 + this.mon*1000000 + (this.year-2000)*100000000;
    }
    getYear(){
        return this.year;
    }
    getMonth(){
        return this.getMonth;
    }
    getDay(){
        return this.day;
    }
    getHour(){
        return this.hour;
    }
    getMinute(){
        return this.min;
    }
    digital(){
        return this.digital
    }
    getTime(){
        return this.time;
    }
}

File: version1/modules/date_time.py
class DateTime:
    def __init__(self,year,mon,day,hour,mins):
        self.year = year
        self.mon = mon
        self.day = day
        self.hour = hour
        self.min = mins
        self.date = {
            "year": self.year,
            "mon": self.mon,
            "day": self.day,
            "hour": self.hour,
            "min":self.min
        }
        self.digital = 0

    def set_date(self,year,mon,day,hour,min):
        self.year = year
        self.mon = mon
        self.day = day
        self.hour = hour
        self.min = mins
        self.date = {
            "year": self.year,
            "mon": self.mon,
            "day": self.day,
            "hour": self.hour,
            "min":self.min
        }
        self.digital = 0

    # trả về date time theo kiểu dictionary
    def get_date(self):
        return self.date

    def digitized(self):
        self.digital = self.min*self.hour*self.day*self.mon *(self.year-2000)
        
    def digital_time(self):
        if self.digital == 0:
            self.digitized()
            return self.digital
        else:
            return self.digital